SonarQube.TeamBuild.PreProcessor/TeamBuildPreProcessor.cs

```diff
@@ -14,43 +14,36 @@
 
 namespace SonarQube.TeamBuild.PreProcessor
 {
-    // was internal
     public class TeamBuildPreProcessor
     {
         public const string FxCopCSharpRuleset = "SonarQubeFxCop-cs.ruleset";
         public const string FxCopVBNetRuleset = "SonarQubeFxCop-vbnet.ruleset";
 
-        private readonly IPropertiesFetcher propertiesFetcher;
-        private readonly IRulesetGenerator rulesetGenerator;
+        private readonly ISonarQubeServerFactory serverFactory;
         private readonly ITargetsInstaller targetInstaller;
 
         #region Constructor(s)
 
         public TeamBuildPreProcessor()
-            : this(new PropertiesFetcher(), new RulesetGenerator(), new TargetsInstaller())
+            : this(new SonarQubeServerFactory(), new TargetsInstaller())
         {
         }
 
         /// <summary>
         /// Internal constructor for testing
         /// </summary>
-        public TeamBuildPreProcessor(IPropertiesFetcher propertiesFetcher, IRulesetGenerator rulesetGenerator, ITargetsInstaller targetInstaller) // was internal
+        public TeamBuildPreProcessor(ISonarQubeServerFactory serverFactory, ITargetsInstaller targetInstaller)
         {
-            if (propertiesFetcher == null)
+            if (serverFactory == null)
             {
-                throw new ArgumentNullException("propertiesFetcher");
-            }
-            if (rulesetGenerator == null)
-            {
-                throw new ArgumentNullException("rulesetGenerator");
+                throw new ArgumentNullException("serverFactory");
             }
             if (targetInstaller == null)
             {
                 throw new ArgumentNullException("rulesetGenerator");
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Method 'TeamBuildPreProcessor.TeamBuildPreProcessor(ISonarQubeServerFactory, ITargetsInstaller)' passes 'rulesetGenerator' as the 'paramName' argument to a 'ArgumentNullException' constructor. Replace this argument with one of the method's parameter names. Note that the provided parameter name should have the exact casing as declared on the method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=fxcop%3AInstantiateArgumentExceptionsCorrectly)